packages/embed-components/src/GlobalStore/sagas.ts

```diff
@@ -0,0 +1,46 @@
+/*
+
+ MIT License
+
+ Copyright (c) 2023 Looker Data Sciences, Inc.
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+
+ */
+import { takeEvery, put } from 'typed-redux-saga'
+import { createFactory } from '@looker/embed-services'
+import type { PayloadAction } from '@reduxjs/toolkit'
+
+import { factoryActions } from './slice'
+import type { InitFactoryAction } from './slice'
+
+function* initSaga(action: PayloadAction<InitFactoryAction>) {
+  const { initFactorySuccessAction, setFailureAction } = factoryActions
```

Comment:
hmm, if I remember correctly I tried this when writing this and it failed because `factoryActions` got destructured before the slice was built. I'll try again.

Reply:
Yep, I get this:
```
    TypeError: Cannot destructure property 'initAction' of '_slice.themeActions' as it is undefined.

      36 |
      37 | const {
    > 38 |   initAction,
         |   ^
      39 |   loadThemeDataAction,
      40 |   getThemesAction,
      41 |   getDefaultThemeAction
```

Reply:
I'm not seeing this when I try it in GlobalStore

Reply:
Not seeing it in theme sagas either